src/main/java/sirius/pasta/noodle/macros/Base64ResourceMacro.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.pasta.noodle.macros;
+
+import org.serversass.ast.FunctionCall;
+import parsii.tokenizer.Position;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Register;
+import sirius.pasta.noodle.Environment;
+import sirius.pasta.noodle.compiler.CompilationContext;
+import sirius.pasta.noodle.compiler.ir.Node;
+import sirius.web.dispatch.SassFunction;
+import sirius.web.http.MimeHelper;
+import sirius.web.resources.Resource;
+import sirius.web.resources.Resources;
+
+import javax.annotation.Nonnull;
+import java.util.Base64;
+import java.util.List;
+
+/**
+ * Provides a macro which encodes the given resource as Base64 string.
+ * <p>
+ * This can be used in CSS like <tt>background: url(base64Resource('/assets/img.png'));</tt> or in IMG tags via
+ * <tt>&lt;img src="@base64Resource('/assets/img.png')" /&gt;</tt>.
+ */
+@Register
+public class Base64ResourceMacro extends BasicMacro implements SassFunction {
+
+    @Part
+    private Resources resources;
+
+    @Override
+    protected Class<?> getType() {
+        return String.class;
+    }
+
+    @Override
+    protected void verifyArguments(CompilationContext compilationContext, Position position, List<Class<?>> args) {
+        if (args.size() != 1 || !CompilationContext.isAssignableTo(args.get(0), String.class)) {
```

Comment:
Warum muss denn die Size der Liste genau 1 sein? Wieso wird dann eine Liste übergeben?

Reply:
Ein Makro bekommt immer die Liste der Argumente die beim Aufruf an sie übergeben wurde (das ist ja generisch in Tagliatelle), in dem Fall erlaubt das Macro halt nur einen Parameter vom Typ String, bei allem anderen meckern wir

Reply:
Ah, okay verstehe